File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;
class Solution {
    public int solution(String s) {
        int answer = 0;
        
        int k = s.length();
        // System.out.println(isCurrect(s));
        StringBuilder sb = new StringBuilder(s);
        
        for(int i=0; i<k-1; i++){
            if(isCurrect(sb.toString())){
                answer++;
            }
            // System.out.println(sb.charAt(0));
            sb.append(sb.charAt(0));
            sb.deleteCharAt(0);
            // System.out.println(sb);
        }
        
        return answer;
    }
    public boolean isCurrect(String bracket){
        Deque<Character> bracketStack = new ArrayDeque<>();
        
        for (int i=0; i<bracket.length(); i++){
            if(bracket.charAt(i)=='('){
                bracketStack.push(')');
            }else if(bracket.charAt(i)=='['){
                bracketStack.push(']');                
            }else if(bracket.charAt(i)=='{'){
                bracketStack.push('}');
            }else{
                if(bracketStack.isEmpty()||bracketStack.pop()!=bracket.charAt(i)){
                    return false;
                }
            }
        }
        if(bracketStack.isEmpty()){
            return true;
        }
        return false;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    var answer = 0;
    
    let sClone = s;
    
    for(let i=0; i<s.length; i++){
        if(isCorrect(sClone)){
            answer++;
        }
        sClone=sClone.substring(1)+sClone.substring(0, 1);
    }
    
    return answer;
}

function isCorrect(bracket){
    const bracketStack = [];
    
    for(let b of bracket){
        if(b==='['){
            bracketStack.push(']')
        }else if(b==='{'){
            bracketStack.push('}')
        }else if(b==='('){
            bracketStack.push(')')
        }else{
            if(bracketStack.length>0){
                if(b===bracketStack[bracketStack.length-1]){
                    bracketStack.pop();
                }
            }else{
                return false;
            }
        }
    }
    if(bracketStack.length===0) {
        return true;
    }
    return false;
}